pytube/extract.py

```diff
@@ -167,20 +167,12 @@ def channel_name(url: str) -> str:
     :returns:
         YouTube channel name.
     """
-    patterns = [
-        r"(?:\/(c)\/([%\d\w_\-]+)(\/.*)?)",
-        r"(?:\/(channel)\/([%\w\d_\-]+)(\/.*)?)",
-        r"(?:\/(u)\/([%\d\w_\-]+)(\/.*)?)",
-        r"(?:\/(user)\/([%\w\d_\-]+)(\/.*)?)"
```

Comment:
can we keep the structure of breaking out the patterns.

Makes it much easier to see what categories of URLs are supported (and is easier to debug) -- versus the combined regex below)